packages/colorpickers/src/elements/ColorPicker/reducer.ts

```diff
@@ -18,12 +19,57 @@ type ColorPickerActionTypes =
   | { type: 'RED_CHANGE'; payload: string }
   | { type: 'GREEN_CHANGE'; payload: string }
   | { type: 'BLUE_CHANGE'; payload: string }
-  | { type: 'ALPHA_CHANGE'; payload: string };
+  | { type: 'ALPHA_CHANGE'; payload: string }
+  | { type: 'RESET_COLOR'; payload: IColor };
 
-type ReducerType = (state: IColor, action: ColorPickerActionTypes) => IColor;
+interface IColorPickerState {
+  color: IColor;
+  hexInput: string;
+  redInput: string;
+  blueInput: string;
+  greenInput: string;
+  alphaInput: string;
+}
+
+export function convertStringToColor(colorString: string): IColor | undefined {
+  if (colorString.includes('#') && !isValidHex(colorString)) {
+    return undefined;
+  }
+
+  const { hue, saturation, lightness } = parseToHsl(colorString);
+  const { red, green, blue, alpha } = parseToRgb(colorString) as IRGBColor;
+  const computedAlpha = alpha === undefined ? 100 : alpha * 100;
+  const computedHex = rgbToHex(red, green, blue);
+
+  return {
+    hue,
+    saturation: saturation * 100,
+    lightness: lightness * 100,
+    red,
+    green,
+    blue,
+    alpha: computedAlpha,
+    hex: computedHex
+  };
+}
+
+export const areColorsEqual = (a: IColor | string | undefined, b: IColor | string | undefined) => {
+  if (a === undefined || b === undefined) {
+    return false;
+  }
+
+  const colorA = typeof a === 'string' ? convertStringToColor(a) : a;
+  const colorB = typeof b === 'string' ? convertStringToColor(b) : b;
 
-export function toState(color?: string | IColor) {
-  const white = {
+  if (colorA === undefined || colorB === undefined) {
+    return false;
+  }
+
+  return isEqual(colorA, colorB);
```

Comment:
`Object.is()` equivalency doesn't cut it within our effects. We have to perform a shallow comparison against these values.